src/components/Visualization.test.tsx

```diff
@@ -41,4 +42,25 @@ describe('Visualization.tsx', () => {
     const element = screen.getByTestId('react-flow-wrapper');
     expect(element).toBeInTheDocument();
   });
+
+  test('should expands the details panel upon clicking on a step', async () => {
+    useVisualizationStore.setState(stepsStub as unknown as RFState);
+    useIntegrationJsonStore.setState(integrationJSONStub as unknown as IIntegrationJsonStore);
```

Comment:
@kahboom I tried to add another test but this statement seems that is not working as expected, so I couldn't manage to select a step and set the information within the store :cry:

have you had the chance to mock the store in the past by any chance?